js/exportsms.js

```diff
@@ -93,79 +111,98 @@ var exportsms = function() {
   }
 
   /**
-   * Process data from a story config passage to the structure expected for the SMS game config.
+   * Get passage data out of passage elements and return as an object with just the info we need.
    *
-   * @param attrs NamedNodeMap of attributes from a DOM element
+   * @param passage
+   *   Passage data as an HTML tw-passagedata element
    * @return Object
    */
-  function _compileStoryConfig(attrs) {
-    var pos,
-        strPos,
-        data = {};
-
-    data.__comments               = attrs.getNamedItem('description') ? attrs.getNamedItem('description').value : '';
-    data.alpha_wait_oip           = attrs.getNamedItem('alpha_wait_oip') ? attrs.getNamedItem('alpha_wait_oip').value : 0;
-    data.alpha_start_ask_oip      = attrs.getNamedItem('alpha_start_ask_oip') ? attrs.getNamedItem('alpha_start_ask_oip').value : 0;
-    data.beta_join_ask_oip        = attrs.getNamedItem('beta_join_ask_oip') ? attrs.getNamedItem('beta_join_ask_oip').value : 0;
-    data.beta_wait_oip            = attrs.getNamedItem('beta_wait_oip') ? attrs.getNamedItem('beta_wait_oip').value : 0;
-    data.game_in_progress_oip     = attrs.getNamedItem('game_in_progress_oip') ? attrs.getNamedItem('game_in_progress_oip').value : 0;
-    data.game_ended_from_exit_oip = attrs.getNamedItem('game_ended_from_exit_oip') ? attrs.getNamedItem('game_ended_from_exit_oip').value : 0;
-    data.story_start_oip          = attrs.getNamedItem('story_start_oip') ? attrs.getNamedItem('story_start_oip').value : 0;
-    data.ask_solo_play            = attrs.getNamedItem('ask_solo_play') ? attrs.getNamedItem('ask_solo_play').value : 0;
-    data.mobile_create = {};
-    data.mobile_create.ask_beta_1_oip         = attrs.getNamedItem('mc_ask_beta_1_oip') ? attrs.getNamedItem('mc_ask_beta_1_oip').value : 0;
-    data.mobile_create.ask_beta_2_oip         = attrs.getNamedItem('mc_ask_beta_2_oip') ? attrs.getNamedItem('mc_ask_beta_2_oip').value : 0;
-    data.mobile_create.invalid_mobile_oip     = attrs.getNamedItem('mc_invalid_mobile_oip') ? attrs.getNamedItem('mc_invalid_mobile_oip').value : 0;
-    data.mobile_create.not_enough_players_oip = attrs.getNamedItem('mc_not_enough_players_oip') ? attrs.getNamedItem('mc_not_enough_players_oip').value : 0;
+  function _extractHTMLData(passage) {
+    var data = {}
+      , attrs = passage.attributes
+      , i
+      , pos
+      ; 
+
+    for (i = 0; i < attrs.length; i ++) {
+      var attrName = attrs[i].name || 'undefinedAttribute' + i;
+      data[attrName] = attrs[i].value || 0;
+    }
 
-    strPos = attrs.getNamedItem('position') ? attrs.getNamedItem('position').value : '0,0';
-    pos = _getPositionFromString(strPos);
+    data.text = passage.innerText.trim(); // Is this still necessary? 
```

Comment:
This may not be necessary. The text is already getting saved in the `_twinedata` object. That one we'll at least want to keep around to help with imports.